### PR TITLE
[SPARK-44558][CONNECT][PYTHON] Export Spark Log Level

### PR DESCRIPTION
**Pull request by cdkrot**
### What changes were proposed in this pull request?

Export Spark Connect Log Level in pyspark. 

### Why are the changes needed?

This is convenient for software dependent on spark-connect so it can be possible to enable debug logging just in one place.

### Does this PR introduce _any_ user-facing change?

New api is suggested.

### How was this patch tested?

Checked it works from shell

### CODE REVIEW

File: python/pyspark/sql/connect/client/core.py [hunk 1]
```diff
@@ -114,6 +115,19 @@ def _configure_logging() -> logging.Logger:
...
+    """
+    This returns this log level as integer, or none (if no logging is enabled).
+
+    Spark Connect logging can be configured with environment variable 'SPARK_CONNECT_LOG_LEVEL'
+    .. versionadded:: 3.5.0
+    """
+
+    if logger.disabled:
+        return None
+    return logger.level
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/client/core.py:128]:
```suggestion
    if not logger.disabled:
        return logger.level
```

File: python/pyspark/sql/connect/client/core.py [hunk 2]
```diff
@@ -114,6 +115,19 @@ def _configure_logging() -> logging.Logger:
 logger = _configure_logging()
 
 
+def spark_connect_log_level() -> Optional[int]:
+    """
+    This returns this log level as integer, or none (if no logging is enabled).
+
+    Spark Connect logging can be configured with environment variable 'SPARK_CONNECT_LOG_LEVEL'
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/client/core.py:122]:
```suggestion
    Spark Connect logging can be configured with environment variable 'SPARK_CONNECT_LOG_LEVEL'

```

File: python/pyspark/sql/connect/client/core.py [hunk 3]
```diff
@@ -114,6 +115,19 @@ def _configure_logging() -> logging.Logger:
 logger = _configure_logging()
 
 
+def spark_connect_log_level() -> Optional[int]:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/client/core.py:118]:
```suggestion
def getLogLevel() -> Optional[int]:
```

File: python/pyspark/sql/connect/client/core.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 __all__ = [
     "ChannelBuilder",
     "SparkConnectClient",
+    "spark_connect_log_level",
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/client/core.py:20]:
```suggestion
    "getLogLevel",
```

**Review by HyukjinKwon — APPROVED** [review]
Looks good otherwise

File: python/pyspark/sql/connect/client/core.py (hunk 1 above)
Comment (cdkrot) [reply python/pyspark/sql/connect/client/core.py]: Need return statement for codestyle:

```
annotations failed mypy checks:
python/pyspark/sql/connect/client/core.py:118: error: Missing return statement  [return]

```

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.

**Pull request closed without merging**